uaclient/util.py

```diff
@@ -497,6 +497,54 @@ def configure_web_proxy(
         request.install_opener(opener)
 
 
+def configure_snap_proxy(
```

Comment:
This probably should live in uaclient.entitlements.livepatch instead of uaclient.util because it really only applies to livepatch setup and config.

Reply:
Currently, this is true. But when we have the `ua refresh config` logic we will be using this function outside the scope of livepatch. That's why I think we should keep it in `util`

Reply:
Per your  comment, about  `ua refresh config`  being outside the scope of livepatch, will be doing comparable with calling `apt.setup_apt_proxy` 

Sorry, maybe my original comment was unclear; I meant a keeping this separate utility function or function(s) called `configure_snap_proxy` or configure_livepatch_proxy` not a LivepatchEntitlement class method. The reason I suggest having the functions housed in a more specific module is because adding a bunch of unrelated things in uaclient/util.py eventually makes that module harder to maintain because it becomes too generic and could grow any function we don't have a more appropriate home for.

That said, I don't feel super strongly about this because snap + livepatch config manipulation is a bit broader than just livepatch.  In the case of apt proxy, we did move the `setup_apt_proxy` function into the apt.py module because we want all things "apt" related in one place and it's a clear cut bit of APT-only functionality.

I defer to your perspective on this,  but had wanted to explain a bit more the motivation for my original comment.

Reply:
Oh fair enough. No problem then, I have moved those functions into the `livepatch.py` module